src/hooks/useNestedMenu.jsx

```diff
@@ -0,0 +1,143 @@
+    items.forEach(item => {
+      if (item.children?.length && showChildren[item.name]) {
+        dispatch(actions[`${item.name}Close`]);
+        closeNested(item.children);
+      }
+    });
+  };
+
+  const toggleNested = (item, e) => {
+    e.stopPropagation();
+    const siblings = getSiblings(item.name);
+    closeNested(siblings);
+    if (showChildren[item.name] !== undefined) {
+      dispatch(actions[`${item.name}Open`]);
+    }
+  };
+
+  const handleMenuLeave = (e, openMenu = false, delay = 500) => {
+    e.stopPropagation();
+    openMenu
+      ? debounce(0, () => {})
+      : debounce(delay * 1.5, () => closeNested(menuItems));
+  };
+
+  const openSubmenu = (name, e = null) => {
+    e?.stopPropagation();
+    if (!showChildren[name] && actions[name + 'Open']) {
+      dispatch(actions[name + 'Open']);
+    }
+  };
+
+  const getSiblings = useCallback(
+    (itemName, children = null) => {
+      let siblings = [];
+      let currentChildren = children || menuItems;
+      currentChildren.forEach(item => {
+        if (showChildren[item.name]) {
+          item.children.find(i => i.name === itemName)
+            ? (siblings = item.children)
+            : (siblings = getSiblings(itemName, item.children));
+        }
+      });
+      siblings = siblings.length
+        ? siblings.filter(({ name }) => name !== itemName)
+        : siblings;
+      return siblings;
+    },
+    [menuItems, showChildren]
+  );
+
+  const setIsItemActive = useCallback(
+    (item, parent = null) => {
+      if (!parent) return;
+      if (item.children?.length && menuItems.find(i => i === parent)) {
+        return showChildren[item.name];
+      } else {
+        const siblings = getSiblings(item.name);
+        return !siblings.some(sibling => showChildren[sibling.name]);
+      }
+    },
+    [menuItems, showChildren]
+  );
+
+  const renderMenu = (items = menuItems, isNested = false, parent = null) => {
+    return items.map(item => (
+      <LinkWrapper
+        key={item.name}
+        item={item}
+        isNested={isNested}
+        ref={refs[item.name]}
+        isActive={setIsItemActive(item, parent)}
+        handleEnter={
+          parent ? e => toggleNested(item, e) : e => openSubmenu(item.name, e)
+        }
+        handleClick={
+          !item.children?.length || !parent
+            ? handleMenuLeave
+            : e => toggleNested(item, e)
+        }>
+        {item.children?.length && shouldRender[item.name] && (
+          <Dropdown
+            origin={{
+              top: refs.nav.current.getBoundingClientRect().bottom,
+              left: parent
+                ? refs[item.name]?.current?.getBoundingClientRect().right
+                : refs[item.name]?.current?.getBoundingClientRect().left,
+            }}
+            anchorRef={refs[item.name]}
+            delay={delay}
+            random={Math.random()}
+            shouldMount={showChildren[item.name]}>
+            {renderMenu(item.children, true, item)}
```

Comment:
you will have to create another separate component of it to iterate over the menu items array and render it. This component will accept as a prop whether to show a submenu or not. `MenuItems` name?